### ISSUE TITLE
Fog vs. Metagems recommendation

### ISSUE THREAD

**Issue by Temikus** [fog/fog#3876]
Do we actually plan to continue support of fog/fog as "require everything" gem? I'm asking since some projects still use 'fog' to support only a specific provider (like ManageIQ with google) and this causes long require times and a ton of unnecessary dependencies to be loaded.

If not and we want people to use the metagems directly, maybe we should make a notice of that during install/in README, like:
NOTICE: You shouldn't require 'fog' gem directly except for:
- Supporting Ruby1.9
- Using a provider that doesn't have a metagem.

**Comment by geemus** [comment]
Sounds reasonable. I think we want to continue to do the metagem thing for now for backwards compatibility at least until 2.0. Maybe once we get everything split out the meta-gem can just be retired though and we can just focus on the individual things.

**Comment by connorshea** [comment]
Is it possible right now to just use the specific depencies we need, or is the metagem necessary until 2.0 is released? I was looking into how to remove extra depencies in GitLab and fog seems like a good place to start.

**Comment by Temikus** [comment]
@connorshea AFAIK as long as your provider is separated to a metagem and you don't need to support Ruby1.9.3, you **should** be using the specific metagem you need.

This is precisely what I was talking about...

@geemus - how about a `spec.post_install_message`? 

With something like: 
```
------------------------------
Thank you for installing fog!

IMPORTANT NOTICE:
If there's a metagem available for your cloud provider, e.g. `fog-aws`,
you should be using it instead of requiring the full fog collection to avoid 
unnecessary dependencies.

'fog' should be required explicitly only if:
- The provider you use doesn't yet have a metagem available.
- You require Ruby 1.9.3 support.
------------------------------
```

**Comment by connorshea** [comment]
@Temikus [Yeah...](https://github.com/gitlabhq/gitlabhq/blob/master/Gemfile.lock#L219)

**Comment by plribeiro3000** [comment]
@Temikus You have it already then. Would you have time to send a PR with that since is your idea?

**Comment by Temikus** [comment]
@connorshea Looking diagonally through the code it looks to me like you have only AWS to worry about: [1](https://github.com/gitlabhq/gitlabhq/blob/91fa250038e9182988319f088fb84741b6e2efc9/config/initializers/carrierwave.rb),[2](https://github.com/gitlabhq/gitlabhq/blob/95c7aa62defdbc7984c71faeb0b901a7fe59b223/doc/raketasks/backup_restore.md) I would double check to make sure though ;)

@plribeiro3000 Sure, I'll make one now.

**Comment by geemus** [comment]
Sounds great, looks like this is closed by #3877

**Comment by connorshea** [comment]
@Temikus thanks for looking into that for me :D I've opened an issue in GitLab so hopefully it'll be improved soon.
